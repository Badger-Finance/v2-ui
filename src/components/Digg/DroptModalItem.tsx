import { Grid, Typography, Button, makeStyles } from '@material-ui/core';
import BigNumber from 'bignumber.js';
import { observer } from 'mobx-react-lite';
import { StoreContext } from 'mobx/store-context';
import React, { useContext } from 'react';

export interface DroptModalItemProps {
	token: string;
	balance: BigNumber;
	displayBalance: string;
	redemptionAmount: string;
	redemptionContract: string;
}

const useStyles = makeStyles((theme) => ({
	droptPaper: {
		padding: theme.spacing(2),
		marginBottom: theme.spacing(2),
		textAlign: 'center',
		minWidth: '15%',
	},
	droptItem: {
		marginBottom: theme.spacing(1),
		paddingLeft: theme.spacing(0.5),
	},
	droptTitleText: {
		marginBottom: theme.spacing(1),
	},
	negativeTopMargin: {
		marginTop: -theme.spacing(0.5),
	},
	droptModalButton: {
		marginBottom: theme.spacing(1),
	},
	modal: {
		display: 'flex',
		alignItems: 'center',
		justifyContent: 'center',
	},
}));

const DroptModalItem = observer((props: DroptModalItemProps) => {
	const store = useContext(StoreContext);

	const classes = useStyles();
	const { token, balance, displayBalance, redemptionAmount, redemptionContract } = props;
	const { rebase } = store;

	return (
		<Grid container direction="row" alignItems="center">
			<Grid item xs={8}>
				<Grid className={classes.droptItem} container direction="column" alignItems="flex-start">
					<Typography variant="subtitle2" color="textSecondary">
						{token}
					</Typography>
					<Typography className={classes.negativeTopMargin}>{displayBalance}</Typography>
					<Typography className={classes.negativeTopMargin} variant="caption" color="textSecondary">
						({redemptionAmount} bDIGG)
					</Typography>
				</Grid>
			</Grid>
			<Grid item xs={4}>
				<Button
					variant="contained"
					size="small"
					color="primary"
					onClick={async () => await rebase.redeemDropt(redemptionContract, balance)}
				>
					Redeem
				</Button>
			</Grid>
		</Grid>
	);
});

export default DroptModalItem;
