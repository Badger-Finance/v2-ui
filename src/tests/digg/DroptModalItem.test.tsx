import React from 'react';
import '@testing-library/jest-dom';
import DroptModalItem from 'components/Digg/DroptModalItem';
import BigNumber from 'bignumber.js';
import { customRender } from 'tests/Utils';

describe('Dropt Modal Item', () => {
  it('renders correctly', () => {
    const { container } = customRender(
      <DroptModalItem
        key="DROPT-2"
        token="DROPT-2"
        balance={new BigNumber(1)}
        displayBalance={'0.001'}
        redemptionAmount={'0.001'}
        redemptionContract={'0x111111111111111111111111'}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
