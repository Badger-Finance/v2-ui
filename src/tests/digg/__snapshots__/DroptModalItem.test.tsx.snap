// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropt Modal Item renders correctly 1`] = `
<div>
  <div
    class="MuiGrid-root-7 MuiGrid-container-8 MuiGrid-align-items-xs-center-16"
  >
    <div
      class="MuiGrid-root-7 MuiGrid-item-9 MuiGrid-grid-xs-8-49"
    >
      <div
        class="MuiGrid-root-7 makeStyles-droptItem-2 MuiGrid-container-8 MuiGrid-direction-xs-column-11 MuiGrid-align-items-xs-flex-start-17"
      >
        <h6
          class="MuiTypography-root-110 MuiTypography-subtitle2-122 MuiTypography-colorTextSecondary-136"
        >
          DROPT-2
        </h6>
        <p
          class="MuiTypography-root-110 makeStyles-negativeTopMargin-4 MuiTypography-body1-112"
        >
          0.001
        </p>
        <span
          class="MuiTypography-root-110 makeStyles-negativeTopMargin-4 MuiTypography-caption-113 MuiTypography-colorTextSecondary-136"
        >
          (
          0.001
           bDIGG)
        </span>
      </div>
    </div>
    <div
      class="MuiGrid-root-7 MuiGrid-item-9 MuiGrid-grid-xs-4-45"
    >
      <button
        class="MuiButtonBase-root-169 MuiButton-root-140 MuiButton-contained-148 MuiButton-containedPrimary-149 MuiButton-containedSizeSmall-159 MuiButton-sizeSmall-161"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label-141"
        >
          Redeem
        </span>
        <span
          class="MuiTouchRipple-root-172"
        />
      </button>
    </div>
  </div>
</div>
`;
